### DISCUSSION TITLE
Document#touch is broken in v8.1.6 and v8.1.7

### DISCUSSION THREAD

**Discussion by priceline-rosenfeld** [mongodb/mongoid#5913] [General]
A backported change broke the `touch` method. `field` is no longer `nil`, but an empty string in some cases, which means the line below should be fixed:

https://github.com/mongodb/mongoid/blob/8.1-stable/lib/mongoid/touchable.rb#L42

```ruby
if field.present?
```

**Comment by jamis** [comment]
Thanks for bringing this to our attention. I've created a ticket (here: https://jira.mongodb.org/browse/MONGOID-5833) where we will track this. 👍

**Reply by priceline-rosenfeld** [reply]
Thank you

**Reply by DarshanaVenkatesh** [reply]
Hi @priceline-rosenfeld! What behavior are you seeing when the field is an empty string? We tested touch with an empty field but did not receive any specific errors. Would you be able to send a script that simulates the error you're seeing?

**Reply by priceline-rosenfeld** [reply]
It has been a while since I noticed this problem, so I don't quite remember how I noticed this bug, but I wasn't using touch with any special params. It's just that `field` is nil by default (I didn't change it when calling `touch`), but it becomes empty after this line: `field = database_field_name(field)`.

When there is a parent, this is the current logic:

```ruby
          if field
            # If we are told to also touch a field, perform a separate write
            # for that field. See MONGOID-5136.
            # In theory we should combine the writes, which would require
            # passing the fields to be updated to the parents - MONGOID-5142.
            sets = set_field_atomic_updates(field)
            selector = atomic_selector
            _root.collection.find(selector).update_one(positionally(selector, sets), session: _session)
          end
```

Here's the thing, I haven't asked to touch a particular field, but this block is being executed because field is now an empty string after v8.1.6. It will try to run `set_field_atomic_updates('')`, for example. I know some code was failing to run correctly after upgrading Mongoid to 8.1.6, but it used to run correctly on 8.1.5 and I tracked down that the issue was this change by that time. But I can no longer remember what specific code was failing by that time. We're currently maintaining our own fork of Mongoid with a few fixes in order to being able to upgrade our Mongoid from 7.5.4 to 8.1-stable, and we fixed this line in our fork (among another validation fix for another issue I reported a while ago), so all tests are currently passing.
